### PR TITLE
adding a property to the result function to allow clearing

### PR DESCRIPTION
**Pull request by alexreardon**
I am on the fence about this one. What do you think @thejameskyle ?

### CODE REVIEW

File: test/index.js [hunk 1]
```diff
@@ -483,6 +483,10 @@ describe('memoizeOne', () => {
     });
   });
 
+  describe('clearing the cache', () => {
```
Comment (alexreardon) [inline test/index.js]: I will write the tests if people think it is a good idea to extend the api like this

**Comment by alexreardon** [comment]
The work around is that you dump the old function and create a new memoized function to get one with a fresh cache

**Comment by alexreardon** [comment]
The sad thing about the .reset is that it breaks the idea that you are getting the exact same function back - but just with memoized arguments

**Comment by alexreardon** [comment]
In @atlaskit/drag-and-drop we memoize movements. When a drag ends the move memoization cache is not cleared. So if you start dragging again and the values match the previous move from the last drag then the function will not be fired.

- Drag
- Move (creates cache)
- Move (hits cache)
- Drop

- Drag
- Move (hits cache and does not fire)
- Drop

Ideally it would be this:

- Drag
- Move (creates cache)
- Move (hits cache and does not fire)
- Drop (clears move cache)

- Drag
- Move (creates cache)
- Move (hits cache and does not fire)
- Drop (clears move cache)

**Comment by thejameskyle** [comment]
Wouldn't the arguments passed to the memoized function be different though (causing it to be a cache miss)?

**Comment by treshugart** [comment]
The use case I think was where a new drag happens at the same exact point. I discussed this with Alex IRL and you can get around having to provide more API if you just re-setup the memoized function.

**Comment by alexreardon** [comment]
@thejameskyle not if it is the exact same point

**Comment by thejameskyle** [comment]
I'm not sure why you wouldn't want it to be memoized if its the exact same point, but I'm probably missing context here

**Comment by alexreardon** [comment]
Same point - but for a different item dragging

**Comment by treshugart** [comment]
@alexreardon could an argument to the function be the item being dragged? If it's different then it would still break cache. If it's not needed, though, then it would be kinda hacky.

**Comment by alexreardon** [comment]
@treshugart it is only when you are dragging the same item again - each item already has its own cache. The whole use case is very edge - but it made me think about this use case for memoizeOne

**Comment by treshugart** [comment]
Ok. I don't think I'd add it for a few reasons:

- Like you said IRL, you're only losing one frame and the issue was because tests were only dragging one pixel to the same locations.
- The result, which is being mutated here, is the consumers value. This would be unexpected and could break things.

**Pull request closed without merging**